tests/unknown_endpoints_test.go

```diff
@@ -80,7 +80,5 @@ func TestUnknownEndpoints(t *testing.T) {
 		queryUnknownEndpoint(t, alice, []string{"_matrix", "media", "unknown"})
 		// v3 should exist, but not v3/unknown.
 		queryUnknownEndpoint(t, alice, []string{"_matrix", "media", "v3", "unknown"})
-
-		queryUnknownMethod(t, alice, "PUT", []string{"_matrix", "media", "v3", "upload"})
```

Comment:
This is just an example endpoint, removing this reduces test coverage. Maybe switch it to `PATCH`?